### PR TITLE
Cassie behavior of the spawn and death of a customrole

### PR DESCRIPTION
**Pull request by InziDeveloperMode**
Made with [An4r3w](https://github.com/An4r3w)

### CODE REVIEW

**Comment by An4r3w** [comment]
.

**Comment by An4r3w** [comment]
😁

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 1]
```diff
@@ -145,6 +145,16 @@ public abstract class CustomRole
         /// </summary>
         public virtual Broadcast Broadcast { get; set; } = new Broadcast();
 
+        /// <summary>
+        /// Gets or sets a value indicating Cassie message that will be played when the Custom Role spawns
+        /// </summary>
+        public abstract string SpawnCassie { get; set; }
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:151]:
```suggestion
        public virtual string SpawnCassie { get; set; } = "";
```

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 2]
```diff
@@ -145,6 +145,16 @@ public abstract class CustomRole
...
+
+        /// <summary>
+        /// Gets or sets a value indicating Cassie message that will be played when the Custom Role spawns
+        /// </summary>
+        public abstract string DeathCassie { get; set; }
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:156]:
```suggestion
        public virtual string DeathCassie { get; set; } = "";
```

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 3]
```diff
@@ -901,7 +911,12 @@ private void OnInternalChangingNickname(ChangingNicknameEventArgs ev)
         private void OnInternalSpawning(SpawningEventArgs ev)
         {
             if (!IgnoreSpawnSystem && SpawnChance > 0 && !Check(ev.Player) && ev.Player.Role.Type == Role && Loader.Random.NextDouble() * 100 <= SpawnChance)
+            {
                 AddRole(ev.Player);
+
+                if (SpawnCassie != null)
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:917]:
```suggestion
                if (SpawnCassie != "")
```

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 4]
```diff
@@ -938,5 +953,11 @@ private void OnSpawningRagdoll(SpawningRagdollEventArgs ev)
         }
 
         private void OnDestroying(DestroyingEventArgs ev) => RemoveRole(ev.Player);
+
+        private void OnDying(DyingEventArgs ev)
+        {
+            if(DeathCassie != null && Check(ev.Player))
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:959]:
```suggestion
            if (DeathCassie != "" && Check(ev.Player))
```

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 5]
```diff
@@ -145,6 +145,16 @@ public abstract class CustomRole
         /// </summary>
         public virtual Broadcast Broadcast { get; set; } = new Broadcast();
 
+        /// <summary>
+        /// Gets or sets a value indicating Cassie message that will be played when the Custom Role spawns
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:149]: Rewrite this pls

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 6]
```diff
@@ -145,6 +145,16 @@ public abstract class CustomRole
...
+        /// </summary>
+        public abstract string SpawnCassie { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating Cassie message that will be played when the Custom Role spawns
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:154]: Same as above

**Review by VALERA771 — COMMENTED** [review]
Do you assign dying event? And better it to be Died event

**Comment by InziDeveloperMode** [comment]
> Do you assign dying event? And better it to be Died event

Andrew's fault 🤓 
Now I'll fix it

**Comment by InziDeveloperMode** [comment]
@VALERA771 done

**Comment by VALERA771** [comment]
You're still not registering event

**Comment by InziDeveloperMode** [comment]
> You're still not registering event

uuh, sorry, I forgot, I fixed it

**Comment by InziDeveloperMode** [comment]
fuck, I accidentally messed everything up. I hate GitHub. I try to fix it.

**Comment by InziDeveloperMode** [comment]
fixed

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 7]
```diff
@@ -145,6 +145,16 @@ public abstract class CustomRole
         /// </summary>
         public virtual Broadcast Broadcast { get; set; } = new Broadcast();
 
+        /// <summary>
+        /// Gets or sets a value indicating Cassie message that will be played when the Custom Role death.
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:149]:
```suggestion
        /// Gets or sets a Cassie message that will be played when the Custom Role spawns.
```

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 8]
```diff
@@ -145,6 +145,16 @@ public abstract class CustomRole
...
+        /// </summary>
+        public virtual string SpawnCassie { get; set; } = "";
+
+        /// <summary>
+        /// Gets or sets a value indicating Cassie message that will be played when the Custom Role spawns.
```
Comment (VALERA771) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:154]:
```suggestion
        /// Gets or sets a Cassie message that will be played when the Custom Role owner dies.
```

**Comment by NaoUnderscore** [comment]
Custom APIs modifications are frozen on this branch. Refer to apis-rework branch instead.

**Pull request closed without merging**